src/Development/IDE/Core/Compile.hs

```diff
@@ -342,28 +272,9 @@ parseFileContents
        -> FilePath  -- ^ the filename (for source locations)
        -> Maybe SB.StringBuffer -- ^ Haskell module source text (full Unicode is supported)
        -> ExceptT [FileDiagnostic] m ([FileDiagnostic], ParsedModule)
-parseFileContents preprocessor filename mbContents = do
+parseFileContents sourcePlugin filename mbContents = do
```

Comment:
I don’t think this name is a good idea given that those are not actually source plugins. It would be great if we could make them source plugins at some point but for now something like `customPreprocessor` is probably a better name.